index.bs

```diff
@@ -1215,6 +1215,383 @@ implementation of this step is specific to the user agent.", the controlling
+: payload
+:: The encoded video frame (encoded image).  The codec and codec parameters are
+    inferred from the properties of the encoding.
+
+
+Data {#media-streaming-data}
+------------------------------------
+
+Senders may send timed data to receivers by sending data-frame messages (see
+[[#appendix-a]]) with the following keys and values.  A data frame message
+contains an arbitrary payload that can be synchronized with and video, such as
+text track data.  A series of data frames that share a data type and timeline
+form a data encoding.
+
+To allow for data frames to be sent out of order, they may be sent in separate
+QUIC streams, but more than one data frame may be sent in one QUIC stream if
+that makes sense for a specific type of data.
+
+Text track data uses a payload type of text, a default duration of unknown, and
+a timescale of 1000000 (microseconds).
+
+: encoding-id
+:: Identifies the media encoding to which this data frame belongs.  This can be
+    used to reference properties of the encoding such as the type of data and
+    time scale.  Referencing properties of the encoding through the encoding id
+    helps to avoid sending duplicate information in every frame.
+
+: sequence-number
+:: Identifies the frame and its order in the encoding.
+    Within an encoding, larger sequence numbers mean later start times.
+    Within an encoding, gaps in sequence numbers mean frames are missing.
+
+: start-time
+:: Identifies the beginning of the time range of the data frame.  The time
+    scale is inferred from the properties of the encoding.  The end time can be
+    inferred from the start time and duration.
+
+: duration
+:: If present, the duration of the data frame.  The time
+    scale is inferred from the properties of the encoding.  Likewise, if not
+    present, the duration is inferred from the properties of the encoding.
+
+: sync-time
+:: If present, a time used to synchronize the start time of this audio frame (and
+    thus, this encoding) with that of other media encodings on different
+    timelines.
+
+: payload
+:: The encoded audio samples.  The codec and codec parameters are inferred from
+    the properties of the encoding.
+
+Feedback {#streaming-feedback}
+------------------------------------
+
+The receiver can send feedback to the sender, such as key frame requests.
+
+A video key frame is requested by sending a video-request message with
+the following keys and values.
+
+To allow for video frames to be sent out of order, they may be sent in separate
+QUIC streams.
+
+: encoding-id
+:: The encoding for which the sender should send a new key frame.
+
+: sequence-number
+:: Gives the order in the encoding.
+    Within an encoding, larger sequence numbers invalidate previous ones.
+    A sender may ignore smaller sequence numbers after a larger one has been processed.
+    This it to prevent out-of-order requests from generating more key frames than necessary. 
+
+: highest-decoded-frame-sequence-number: uint
```

Comment:
You can't assume that received frames will be decoded because you won't know about a decode failure until you try and decode, and then you're stuck because you told the sender you received it, but then you weren't able to decode it (or you need to send another message saying you didn't really decode it).  The point of a key frame request is to recover from decode failure.

But this allows for a slight optimization where the sender can diff a previous frame instead of generating a full key frame.   I'm not sure how useful that will be in practice, but even just for debugging, info this might be nice to have.